### PR TITLE
New PR With Minimal Functional Changes and Warnings for Azure Sql Failover Group AutoDR Feature

### PR DESCRIPTION
**Pull request by ziwa-msft**
## Description

This is the new PR that contains only the most mission critical changes for this release:
    - Add warnings
    - fix nullifying logic
    - incorporating Hydra fix for operation pollings
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x ] Title of the pull request is clear and informative.
- [ x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x ] Pull request includes test coverage for the included changes.
- [x ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/ResourceManager/Sql/Commands.Sql/Failover Group/Services/AzureSqlFailoverGroupAdapter.cs [hunk 1]
```diff
@@ -145,20 +145,28 @@ internal AzureSqlFailoverGroupModel UpsertFailoverGroup(AzureSqlFailoverGroupMod
...
+            readOnlyEndpoint.FailoverPolicy = model.ReadOnlyFailoverPolicy == null ? AllowReadOnlyFailoverToPrimary.Disabled.ToString() : model.ReadOnlyFailoverPolicy;
 
             ReadWriteEndpoint readWriteEndpoint = new ReadWriteEndpoint();
-            readWriteEndpoint.FailoverPolicy = model.ReadWriteFailoverPolicy;
+            readWriteEndpoint.FailoverPolicy = model.ReadWriteFailoverPolicy == null ? FailoverPolicy.Manual.ToString() : model.ReadWriteFailoverPolicy;
 
-            if (model.FailoverWithDataLossGracePeriodHours.HasValue)
+            if (!string.Equals(model.ReadWriteFailoverPolicy, FailoverPolicy.Manual.ToString()))
             {
-                readWriteEndpoint.FailoverWithDataLossGracePeriodMinutes = model.FailoverWithDataLossGracePeriodHours * 60;
```
Comment (dlemMSFT) [inline src/ResourceManager/Sql/Commands.Sql/Failover Group/Services/AzureSqlFailoverGroupAdapter.cs]: This is not needed -- can you undo?

File: src/ResourceManager/Sql/Commands.Sql/Failover Group/Services/AzureSqlFailoverGroupAdapter.cs [hunk 2]
```diff
@@ -107,11 +107,11 @@ internal AzureSqlFailoverGroupModel UpsertFailoverGroup(AzureSqlFailoverGroupMod
             partnerServers.Add(partnerServer);
 
             ReadOnlyEndpoint readOnlyEndpoint = new ReadOnlyEndpoint();
-            readOnlyEndpoint.FailoverPolicy = model.ReadOnlyFailoverPolicy;
+            readOnlyEndpoint.FailoverPolicy = model.ReadOnlyFailoverPolicy == null ? AllowReadOnlyFailoverToPrimary.Disabled.ToString() : model.ReadOnlyFailoverPolicy;
```
Comment (dlemMSFT) [inline src/ResourceManager/Sql/Commands.Sql/Failover Group/Services/AzureSqlFailoverGroupAdapter.cs]: This change and the other 3 symmetric do not actually influence behavior and are not needed

File: src/ResourceManager/Sql/Commands.Sql/Failover Group/Cmdlet/RemoveAzureSqlFailoverGroup.cs [hunk 3]
```diff
@@ -39,6 +40,7 @@ public class RemoveAzureSqlFailoverGroup : AzureSqlFailoverGroupCmdletBase
         [Parameter(Mandatory = false,
             HelpMessage = "The partner resource group name for Azure SQL Database Failover Group.")]
         [ValidateNotNullOrEmpty]
+        [Obsolete("This parameter will be deprecated in the next release.")]
```
Comment (dlemMSFT) [inline src/ResourceManager/Sql/Commands.Sql/Failover Group/Cmdlet/RemoveAzureSqlFailoverGroup.cs]: I think you mean "the next major version change"

File: src/ResourceManager/Sql/ChangeLog.md [hunk 4]
```diff
@@ -18,7 +18,10 @@
         - Additional information about change #1
 -->
 ## Current Release
-
+* Bug fixes on Azure Failover Group Cmdlets
+	- Corporate Hydra fix for operation polling
```
Comment (jaredmoo) [inline src/ResourceManager/Sql/ChangeLog.md]: Corporate? Hydra? I suggest something like "Fix for operation polling"

Comment (jaredmoo) [inline src/ResourceManager/Sql/ChangeLog.md]: Imagine you're a powershell customer. They don't know what hydra is.

**Pull request merged**